server/core/src/https/v1.rs

```diff
@@ -1396,6 +1396,14 @@ pub async fn credential_update_update(
             return Err(WebError::InternalServerError(errmsg));
         }
     };
+
+    if matches!(scr, CURequest::TotpVerify(_, ref label) if label.trim().is_empty()) {
+        // I'd really rather have a 400 here. Do I add a WebError::HttpError(u16, String)?
```

Comment:
You're trying to double validate here then, that's the issue and why it feels "off". Because the validation oef the data is actually done in the idmcredentuialupdate which is better positioned to serve a contextual and informed response.

So remove this block, it's not needed.

Reply:
It is double validating. **But** that is for additional contextual information in terms of Http response. We should be returning an invalid input early as 400, Yet currently a "InvalidName" is returned as a 200 json response.

I'd be nice if we made it a little more complaint to how HTTP expects it to work.

Reply:
Happy enough with that. And given that is already the precedent its not worth trying to change now. 

Given I have moved the empty check further in anyways this is no longer the only place checking it